nucleus/hk2/hk2-config/src/main/java/org/jvnet/hk2/config/WriteableView.java

```diff
@@ -394,43 +396,8 @@ else if ( s.startsWith("<") ) {
                 }
                 appliedChanges.add(event);
             }
-            for (ProtectedList entry :  changedCollections.values())  {
-                List<Object> originalList = entry.readOnly;
-                for (PropertyChangeEvent event : entry.changeEvents) {
-                    if (event.getOldValue()==null) {
-                        originalList.add(event.getNewValue());
-                    } else {
-                        final Object toBeRemovedObj = event.getOldValue();
-                        if ( toBeRemovedObj instanceof ConfigBeanProxy ) {
-                            final Dom toBeRemoved = Dom.unwrap((ConfigBeanProxy)toBeRemovedObj);
-                            for (int index=0;index<originalList.size();index++) {
-                                Object element = originalList.get(index);
-                                Dom dom = Dom.unwrap((ConfigBeanProxy) element);
-                                if (dom==toBeRemoved) {
-                                    Object newValue = event.getNewValue();
-                                    if (newValue == null) {
-                                        originalList.remove(index);
-                                    } else {
-                                        originalList.set(index, newValue);
-                                    }
-                                }
-                            }
-                        }
-                        else if ( toBeRemovedObj instanceof String ) {
-                            final String toBeRemoved = (String)toBeRemovedObj;
-                            for (int index=0;index<originalList.size();index++) {
-                                final String item = (String)originalList.get(index);
-                                if (item.equals(toBeRemoved)) {
-                                    originalList.remove(index);
-                                }
-                            }
-                        }
-                        else {
-                              throw new IllegalArgumentException();
-                        }
-                    }
-                    appliedChanges.add(event);
-                }
+            for (ProtectedList<?> entry :  changedCollections.values())  {
+                commitListChanges(entry, appliedChanges);
```

Comment:
NB. The loop body got extracted to the `commitListChanges` method because I needed a bindable type variable to use generics.